plugin/src/generators/lib-tags/generator.ts

```diff
@@ -0,0 +1,198 @@
+import { readJson, Tree, writeJson, getProjects, ProjectConfiguration, formatFiles } from '@nx/devkit';
+import { LibTagsGeneratorSchema } from './schema';
+import { execSync } from 'child_process';
+import { addNxAppTag, addNxScopeTag, askQuestion, getNxRules, getNxRulesEntry, readESLintConfig } from '../../shared/utils';
+import { isEmpty } from 'lodash';
+
+export async function libTagsGenerator(
+  tree: Tree,
+  options: LibTagsGeneratorSchema,
+) {
+  const hasUnstagedChanges = execSync('git status -s').toString('utf8');
+
+  if (hasUnstagedChanges) {
+    const shouldContinue = await askQuestion('You have unstaged changes. Are you sure you want to continue? (y/n): ') === 'y';
+
+    if (!shouldContinue) {
+      return;
+    }
+  }
+
+  const log = (message: string): void => {
+    if (!options.silent) {
+      console.log(message);
+    }
+  };
+
+  // #1 Check eslint config nx-boundaries rule
```

Comment:
Let's move the code for checking eslint config to a separate function

Reply:
done